gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/ActiveClusterMonitor.java

```diff
@@ -134,6 +161,61 @@ private String queryCluster(String target) {
   }
 
   private ClusterStats getPrestoClusterStats(ProxyBackendConfiguration backend) {
+    if (isUseJwt) {
+      return getPrestoClusterStatsSql(backend);
+    }
+    return getPrestoClusterStatsUi(backend);
+  }
+
+  private ClusterStats getPrestoClusterStatsSql(ProxyBackendConfiguration backend) {
+    ClusterStats clusterStats = new ClusterStats();
+    clusterStats.setClusterId(backend.getName());
+    String jdbcUrl;
+    try {
+      jdbcUrl = String.format("jdbc:trino://%s:%s/system/runtime",
+              (new URL(backend.getProxyTo())).getHost(), jdbcPort);
+    } catch (MalformedURLException e) {
+      log.debug("Cannot construct URL from " + backend.getProxyTo());
+      clusterStats.setHealthy(false);
+      return clusterStats;
+    }
+    Properties connectionProperties = new Properties();
+    if (isUseJwt) {
+      connectionProperties.setProperty("accessToken", jwt);
+    }
+    connectionProperties.setProperty("SSL", Boolean.toString(jdbcUseSsl));
+
+    try (Connection conn = driver.connect(jdbcUrl, connectionProperties)) {
+      Statement statement = conn.createStatement();
+      String queryStatsSql = "select\n"
+              + "  count_if(upper(state) = 'RUNNING') as runningQueries,\n"
+              + "  count_if(upper(state) = 'QUEUED') as queuedQueries,\n"
+              + "  count_if(upper(state) = 'BLOCKED') as blockedQueries\n"
+              + "from system.runtime.queries";
+      ResultSet queryStatsResult = statement.executeQuery(queryStatsSql);
+      queryStatsResult.next();
+      clusterStats.setQueuedQueryCount(queryStatsResult.getInt("queuedQueries"));
+      clusterStats.setRunningQueryCount(queryStatsResult.getInt("runningQueries"));
+      clusterStats.setBlockedQueryCount(queryStatsResult.getInt("blockedQueries"));
+
+      String nodeStatsSql = "select\n"
+              + "count_if( not coordinator AND upper(state) = 'ACTIVE') as activeWorkers\n"
+              + "from system.runtime.nodes";
+      ResultSet nodeStatsResult = statement.executeQuery(nodeStatsSql);
+      nodeStatsResult.next();
+      clusterStats.setNumWorkerNodes(nodeStatsResult.getInt("activeWorkers"));
+      clusterStats.setProxyTo(backend.getProxyTo());
+      clusterStats.setExternalUrl(backend.getExternalUrl());
+      clusterStats.setRoutingGroup(backend.getRoutingGroup());
+      clusterStats.setHealthy(true);
+    } catch (SQLException e) {
+      log.error("Error querying cluster stats: " + e);
+    }
+    clusterStats.setHealthy(false);
```

Comment:
Intended to set it to false here no matter what?

Reply:
Maybe this is intended to be inside the catch

Reply:
same as above - forgot a return statement in the try block, so this should only be reached if the health check fails.